app/src/main/java/com/example/tasky/feature_authentication/presentation/login_screen/LoginScreen.kt

```diff
@@ -0,0 +1,182 @@
+package com.example.tasky.feature_authentication.presentation.login_screen
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import com.example.tasky.R
+import com.example.tasky.feature_authentication.domain.util.AuthUseCaseResult
+import com.example.tasky.feature_authentication.domain.validation.EmailError
+import com.example.tasky.feature_authentication.domain.validation.PasswordError
+import com.example.tasky.feature_authentication.presentation.components.TextFieldComponent
+import com.example.tasky.presentation.theme.BackgroundBlack
+import com.example.tasky.presentation.theme.BackgroundWhite
+import com.example.tasky.presentation.theme.BtnNavRegScreen
+import com.example.tasky.presentation.theme.LoginBtnTextColor
+import com.example.tasky.presentation.theme.HintColor
+import com.example.tasky.util.Screen
+
+@Composable
+fun LoginScreen(
+    navController: NavController,
+    viewModel: LoginViewModel = hiltViewModel()
+) {
+    val state by viewModel.state.collectAsState()
```

Comment:
Recently the collectAsStateWithLifecycle() dependency got stable, so I'd recommend to use that since it's the safest way to collect flows in Android (currently, your flow will be collected even if the app is in the background). For that, you'll need this dependency:
```
implementation "androidx.lifecycle:lifecycle-runtime-compose:2.6.0"
```
Then use it like this:
```
val state by viewModel.state.collectAsStateWithLifecycle()
```